File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/multiplex/API/ParseListener.java
package com.lilanz.tooldemo.multiplex.API;

import android.support.annotation.NonNull;

import com.google.gson.Gson;

import java.util.List;

// 监听器：获取解析后的数据
public abstract class ParseListener<T> {
    public void jsonParsed(@NonNull List<T> beanList) {
    }

    public void jsonParsed(T t) {
    }

    // 请求成功后的回调信息
    public void onTip(String msg) {
    }

    @NonNull
    @Override
    public String toString() {
        return new Gson().toJson(this);
    }

    // 回调的错误信息
    public abstract void onError(int errCode, String errMsg);
}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/api/ParseListener.kt
package com.lilanz.kotlintool.api

/**
 * 监听器：获取网络请求解析后的数据
 */
interface ParseListener<T> {
    /**
     * 解析对象列表
     */
    fun jsonParsed(beanList: List<T>) {}

    /**
     * 解析单个对象
     */
    fun jsonParsed(t: T) {}

    /**
     * @param jsonStr 请求结果 json字符串
     */
    fun jsonResult(jsonStr: String?) {}

    /**
     * 请求成功后的回调信息
     */
    fun onTip(msg: String?) {}

    /**
     * 回调的错误信息
     */
    fun onError(errCode: Int, errMsg: String?)
}